rosbag2_storage/src/rosbag2_storage/metadata_io.cpp

```diff
@@ -73,14 +96,42 @@ struct convert<rosbag2_storage::TopicInformation>
     return node;
   }
 
-  static bool decode(const Node & node, rosbag2_storage::TopicInformation & metadata)
+  static bool decode(const Node & node, rosbag2_storage::TopicInformation & metadata, int version)
   {
-    metadata.topic_metadata = node["topic_metadata"].as<rosbag2_storage::TopicMetadata>();
+    metadata.topic_metadata = decode_for_version<rosbag2_storage::TopicMetadata>(
+      node["topic_metadata"], version);
     metadata.message_count = node["message_count"].as<uint64_t>();
     return true;
   }
 };
 
+template<>
+struct convert<std::vector<rosbag2_storage::TopicInformation>>
+{
+  static Node encode(const std::vector<rosbag2_storage::TopicInformation> & rhs)
+  {
+    Node node(NodeType::Sequence);
+    for (auto value : rhs) {
```

Comment:
```suggestion
    for (auto value & : rhs) {
```

Given that this function copies one vector into, what seems like, another vector, can this be constructed in place in the `Node` constructor? Alternatively, I would recommend `reserve`ing the node sequence.
Obviously, please disregard all this if `Node` doesn't allow us to do this.

Reply:
There's a vector but it's behind a few levels of `private`, we don't have access to reserve or iterator-insert, we only have `push_back`.